plugins/flutter_aepedge/android/src/main/java/com/adobe/marketing/mobile/flutter/flutter_aepedge/FlutterAEPEdgeDataBridge.java

```diff
@@ -50,23 +52,36 @@ protected static ExperienceEvent eventFromMap(final Map map) {
              return null;
          }
 
-        Map<String, Object> xdmData = getNullableMap(map, XDM_DATA_KEY);
         String datasetId = null;
+        String datastreamIdOverride = null;
 
+        Map<String, Object> xdmData = getNullableMap(map, XDM_DATA_KEY);
 
-        if (xdmData != null) {
 
+        if (xdmData != null) {
```

Comment:
Refactor this block to exit early in case of null xdm

if(xdmData == null) {
   Log.d(TAG, "eventFromMap() - Cannot create Experience event, xdmData is null.");
   return null
}